### PR TITLE
networkd: allow setting of multicast querier for linux bridge

### LINKED ISSUE #723: allow setting of multicast querier for linux bridge
For Linux bridge, there's a Multicast Querier setting in the sysfs. It is meant as and ”optimisation”, but in my trials it's needed to have stable multicast operation over the bridge.
It is controlled by writing to sysfs, for example:  
`echo "1" > /sys/devices/virtual/net/br0/bridge/multicast_querier`

Could you consider extending Linux bridge configuration in systemd-networkd?  There should be `MulticastQuerier=true/false` option, which if set, should write to sysfs during bridge configuration.

### PR DESCRIPTION
**Pull request by ssahani**
fixes #723

### CODE REVIEW

**Comment by poettering** [comment]
lgtm.

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -310,6 +310,15 @@
             of the Listening and Learning states before the Forwarding state is entered.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>MulticastQuerier=</varname></term>
+          <listitem>
+            <para>A boolean. MulticastQuerier specifies process of listening to Internet Group Management Protocol (IGMP)
+            network traffic. The feature allows a network bridge to listen in on the IGMP conversation between hosts and routers.
+            Defaults to false.
```
Comment (keszybz) [inline man/systemd.netdev.xml]: A boolean. This setting controls the `IFLA_BR_MCAST_QUERIER` option in the kernel. If enabled, the kernel will send general ICMP queries from a zero source address. This feature should allow faster convergence on startup, but it causes some multicast-aware switches to misbehave and disrupt forwarding of multicast packets. If not set, the kernel default (currently "no"), will be preserved.

**Comment by keszybz** [comment]
Please see the suggested man page wording. I think I understand what this setting does, but maybe not, so please double check.

I also think the code should be changed to tri-state: unset, yes, no, with unset being the default, and not override the kernel setting when unset. Seems this is a problematic feature, and the kernel might switch defaults again in the future.

File: man/systemd.netdev.xml (hunk 1 above)
Comment (ssahani) [inline man/systemd.netdev.xml]: Thanks for the man.

**Comment by ssahani** [comment]
@keszybz makes sense we should not touch it . Making this to tristate

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -310,6 +310,17 @@
...
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>MulticastQuerier=</varname></term>
+          <listitem>
+            <para>A boolean. This setting controls the IFLA_BR_MCAST_QUERIER option in the kernel.
+            If enabled, the kernel will send general ICMP queries from a zero source address.
+            This feature should allow faster convergence on startup, but it causes some
+            multicast-aware switches to misbehave and disrupt forwarding of multicast packets.
+            Defaults to unset.
```
Comment (poettering) [inline man/systemd.netdev.xml]: Needs to say that "When unset, the kernel's default setting applies."

**Comment by poettering** [comment]
besides the doc issue i found looks good to me.

**Comment by zonque** [comment]
ftr: the failed ubuntu-amd64 test is a false negative.

**Pull request merged**